Documentation/Common-Gotchas.md

```diff
@@ -0,0 +1,232 @@
+# Common Gotchas
+
+There are some common errors that arise when trying to use Argo. Many are due to
+the fact that Argo is so fundamentally different from how you might be used to
+decoding JSON, and others are due to limitations of Swift. This document aims
+to address some of those situations and set you on the right track!
+
+## `curry` limitations
+
+It is very important to fully curry your model’s `init` in order to use Argo.
+Because of this, `Argo` recommends using
```

Comment:
What do you think about changing this to say "we recommend"? Feels kinda weird to take advice from a library.